README.md

```diff
@@ -1,60 +1,72 @@
+`eslines` helps you to post-process an ESLint JSON report.
+
+The default behavior is transforming all errors into warnings, except 1) those reported in lines modified within the current git branch and 2) any `no-unused-vars` ESLint rule break.
 
 ## How to use it
 
 Install it:
 
 	npm install eslines
 
-Add the default `.eslines.json` config file in you git repo:
-
-	{
-		"rulesNotToDowngrade": [],
-		"remote": "origin/master",
-		"processors": {
-			"default": "lines-modified"
-		}
-	}
+Add the default `.eslines.json` config file to your git repo:
+
+    {
+        "branches": {
+	        "default": ["lines-modified", "enforce"]
+        },
+        "processors": {
+            "lines-modified": {
+                "remote": "origin/master",
+            },
+            "enforce": {
+                "rules": ["no-unused-vars"],
+            }
+        }
+    }
 
 Run it:
 
 	eslint -f json . | eslines
 
 ## Config file
 
-`eslines` reads its configuration from a file named `.eslines.json` placed in the root of your git repository.
-
-`eslines` has two ways of processing an ESLint report, we call them processors. The processor `lines-modified` will downgrade ESLint `errors` to `warnings`, except for lines modified in the current branch. The processor `parsing-errors` takes the ESLint report and outputs a new one containing only `Parsing Errors`.
+`eslines` reads its configuration from a file named `.eslines.json` placed in the root of your git repository. Out of the box, it comes with three ways of post-processing an ESLint report - we call them *processors*.
 
-Options:
-
-* **rulesNotToDowngrade**: an array containing valid [ESLint rule ids](http://eslint.org/docs/rules/). For any rule declared in this array, the `lines-modified` processor won't downgrade to `warnings` reported `errors` for these rules, no matter whether the line was modified or not.
-
-* **remote**: lines modified are determined by diffing this remote git branch against the current branch.
-
-* **processors**: `eslines` will choose the processor to use by default and which one to use for particular branches looking at this object.
+The processor `lines-modified` will transform ESLint `errors` into `warnings`, except for errors that are in lines modified in the current branch. The processor `parsing-errors` takes the ESLint report and outputs a new one containing only `Parsing Errors`. The processor `enforce` will transform any `warning` into an `error`. Processors are composable.
 
 For example:
 
 	{
-		"rulesNotToDowngrade": [],
-		"remote": "origin/master",
+		"branches": {
+			"default": ["lines-modified", "enforce"],
+			"master": ["parsing-errors"],
+			"my/topic-branch": ["parsing-errors"],
+		},
 		"processors": {
-			"default": "lines-modified",
-			"master": "parsing-errors",
-			"my/topic-branch": "parsing-errors"
+			"lines-modified": {
+				"remote": "origin/master",
+			},
+			"enforce": {
+				"rules": ["no-unused-vars"],
```

Comment:
We're creating an abstraction for `enforce` at a time when we have just one use case for it, so it's hard to decide whether it's the right one or not.

At this moment, I think of `enforce` not as a separate processor, but rather as a whitelist of quirky rules for `lines-modified`. Where 'quirky' means that a change on one line can cause a linting error or another.

By the way, that list should also include parsing errors. It's easy to change a line in such a way that there is a parsing error somewhere else. The current linting configuration will fail to catch that (both in precommit hook and on CircleCI) and will report the changes as error-free.

The `lines-modified` and `prettier-formatting` processors have an interesting property: running them in any order doesn't change the result -- they are completely orthogonal. The `enforce` processor is order-dependent. Maybe that's part of the reason why it's not so easy to design properly.